### ISSUE TITLE
Wrong encoding of Swift Date as a query param

### ISSUE THREAD

**Issue by petrpavlik** [soto-project/soto-core#631]
**Describe the bug**
It looks like encoding Swift `Date` as query params of GET requests does not work as expected. Looks like this code https://github.com/soto-project/soto-core/blob/2614aa43ab0eef94b873d84249fc974368b39369/Sources/SotoCore/Encoder/RequestContainer.swift#L66 uses standard description to turn a `Date` into a string which results in an URL looking like this:

```
https://email.us-east-1.amazonaws.com/v2/email/deliverability-dashboard/statistics-report/mail.indiepitcher.com?EndDate=2024-12-29%2023%3A19%3A50%20%2B0000&StartDate=2024-12-29%2023%3A19%3A50%20%2B0000
```

which seems to be rejected by AWS, at least for this particular endpoint. I have a repro when trying to get the sending domain statistics of SES.

**To Reproduce**
```swift
import SotoSESv2

let ses = SESv2(client: awsClient)
let statistics = try await ses.getDomainStatisticsReport(domain: "mail.indiepitcher.com", endDate: Date.now, startDate: Date.now)
```

**Expected behavior**
The request returns `GetDomainStatisticsReportResponse`

**Actual behavior**
The request fails with `SerializationException: '2024-12-29 23:23:53 +0000' can not be converted to Date`

**Setup (please complete the following information):**
 - OS: Ubuntu
 - Version of soto: 7.3.0
 - Authentication mechanism hard-coded credentials

**Comment by adam-fowler** [comment]
Yeah query parameters are by-passing the encoder. Given pretty much all of them are strings and numbers, with some special cases for collections, that doesn't matter. Looks like you found an exception though.

**Comment by adam-fowler** [comment]
#632 should fix this

**Comment by petrpavlik** [comment]
Yes, checked the PR and looks great. Thanks!

**Comment by adam-fowler** [comment]
This has been released in soto-core 7.3.1
